File: Python/Silnia w przedziale.py
def factorial_in_range(a, b):
    n = 0
    count = 0
    factorial = 1
    while factorial <= b:
        if a <= factorial <= b:
            count += 1
        n += 1
        if n > 0:
            factorial *= n
    return count

a, b = map(int, input().split())
print(factorial_in_range(a, b))

File: Java/Silnia w przedziale.java
import java.util.Scanner;

public class Main 
{
    public static int factorialInRange(int a, int b) 
    {
        int n = 0;
        int count = 0;
        long factorial = 1;
        while (factorial <= b) 
        {
            if (a <= factorial && factorial <= b)
                count++;
            n++;
            if(n > 0)
                factorial *= n;
        }
        return count;
    }

    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);
        int a = input.nextInt();
        int b = input.nextInt();
        System.out.print(factorialInRange(a, b));
        input.close();
    }
}